src/main/java/gregtech/common/pipelike/fluidpipe/BlockFluidPipe.java

```diff
@@ -110,35 +110,25 @@ public boolean isHoldingPipe(EntityPlayer player) {
     }
 
     @Override
-    public void onEntityCollision(World worldIn, @Nonnull BlockPos pos, @Nonnull IBlockState state, @Nonnull Entity entityIn) {
+    public void onEntityCollision(@Nonnull World worldIn, @Nonnull BlockPos pos, @Nonnull IBlockState state, @Nonnull Entity entityIn) {
         if (worldIn.isRemote) return;
-        if (entityIn instanceof EntityLivingBase && entityIn.world.getTotalWorldTime() % 20 == 0L) {
+        if (entityIn instanceof EntityLivingBase && worldIn.getTotalWorldTime() % 10 == 0) {
             EntityLivingBase entityLiving = (EntityLivingBase) entityIn;
             TileEntityFluidPipe pipe = (TileEntityFluidPipe) getPipeTileEntity(worldIn, pos);
-            if(!(pipe instanceof TileEntityFluidPipeTickable))
+            if (!(pipe instanceof TileEntityFluidPipeTickable))
                 return;
-            List<Integer> temps = new ArrayList<>();
+
+            // apply temperature damage for each pipe
+            int temperature = 0;
+            int nonEmptyTanks = 0;
             for (FluidTank tank : ((TileEntityFluidPipeTickable) pipe).getFluidTanks()) {
                 if (tank.getFluid() != null && tank.getFluid().amount > 0) {
-                    temps.add(tank.getFluid().getFluid().getTemperature(tank.getFluid()));
+                    temperature += tank.getFluid().getFluid().getTemperature(tank.getFluid());
```

Comment:
Couldn't this average make it so that neither Cyrogenic nor Heat damage were applied if the temperature values were just right? It might be better to sum all the pipes that would apply heat damage and all the pipes that would apply cold damage in separate variables, and then make two calls to `applyTemperatureDamage`, one for each variable.